ocaml/sdk-gen/go/gen_go_binding.ml

```diff
@@ -44,10 +44,59 @@ let render_api_messages_and_errors destdir =
   generate_file ~rendered:messages_rendered ~destdir
     ~output_file:"api_messages.go"
 
+let render_convert_header () =
+  let name s = `O [("name", `String s); ("sname", `Null)] in
+  let obj =
+    `O
+      [
+        ("name", `String "convert")
+      ; ( "modules"
+        , `O
+            [
+              ("import", `Bool true)
+            ; ( "items"
+              , `A (List.map name ["fmt"; "math"; "reflect"; "strconv"; "time"])
+              )
+            ]
+        )
+      ]
+  in
+  render_template "FileHeader.mustache" obj ~newline:true ()
+
+let render_converts destdir =
+  let event = render_template "ConvertBatch.mustache" Convert.event_batch () in
+  let interface =
+    render_template "ConvertInterface.mustache" Convert.interface ()
+  in
+  let param_types = Types.of_params objects in
+  let result_types = Types.of_results objects in
+  let generate types of_json =
+    types
+    |> List.map Convert.of_ty
+    |> List.map (fun params ->
```

Comment:
Can probably perform fusion here to remove the intermediary list.

In the pure setting, `map g (map f xs) = map (g . f) xs` always holds (saving you the redundant intermediary list); in this case, I suspect there is no sequencing of side effects that we must respect (tell me if I'm wrong, as I don't know the semantics of `Convert`!). 

A minor thing, but probably want to push it inwards as the compiler will likely make no effort to:
```ocaml
    types
    |> List.map (fun params ->
           let template = Convert.(template_of_convert (of_ty params)) in
```

Reply:
Nice. I adopted this.